src/Microsoft.AspNetCore.Authentication.Twitter/TwitterHandler.cs

```diff
@@ -148,53 +149,52 @@ public TwitterHandler(IOptionsMonitor<TwitterOptions> options, ILoggerFactory lo
-            canonicalizedRequestBuilder.Append(HttpMethod.Post.Method);
-            canonicalizedRequestBuilder.Append("&");
-            canonicalizedRequestBuilder.Append(UrlEncoder.Encode(RequestTokenEndpoint));
+            var signatureParts = new SortedDictionary<string, string>((queryParameters ?? new Dictionary<string, string>()).Union(authorizationParts).Union(formData ?? new Dictionary<string, string>()).ToDictionary(x => x.Key, x => x.Value));
+            string parameterString = signatureParts.Select(x => string.Format("{0}={1}", Uri.EscapeDataString(x.Key), Uri.EscapeDataString(x.Value))).Aggregate((x, y) => string.Format("{0}&{1}", x, y));
             canonicalizedRequestBuilder.Append("&");
-            canonicalizedRequestBuilder.Append(UrlEncoder.Encode(parameterString));
+            canonicalizedRequestBuilder.Append(Uri.EscapeDataString(parameterString));
 
-            var signature = ComputeSignature(Options.ConsumerSecret, null, canonicalizedRequestBuilder.ToString());
+            var signature = ComputeSignature(Options.ConsumerSecret, accessToken?.TokenSecret, canonicalizedRequestBuilder.ToString());
             authorizationParts.Add("oauth_signature", signature);
 
-            var authorizationHeaderBuilder = new StringBuilder();
-            authorizationHeaderBuilder.Append("OAuth ");
-            foreach (var authorizationPart in authorizationParts)
+            var queryString = (queryParameters != null && queryParameters.Any() ? "?" + queryParameters.Select(x => x.Key + "=" + x.Value).Aggregate((x, y) => x + "&" + y) : "");
+            var request = new HttpRequestMessage(httpMethod, url + queryString);
+            request.Headers.Add("Authorization", "OAuth " + authorizationParts.Select(x => string.Format("{0}=\"{1}\"", x.Key, Uri.EscapeDataString(x.Value))).Aggregate((x, y) => string.Format("{0},{1}", x, y)));
+
+            if (formData != null)
             {
-                authorizationHeaderBuilder.AppendFormat(
-                    "{0}=\"{1}\", ", authorizationPart.Key, UrlEncoder.Encode(authorizationPart.Value));
+                request.Content = new FormUrlEncodedContent(formData);
             }
-            authorizationHeaderBuilder.Length = authorizationHeaderBuilder.Length - 2;
 
-            var request = new HttpRequestMessage(HttpMethod.Post, RequestTokenEndpoint);
-            request.Headers.Add("Authorization", authorizationHeaderBuilder.ToString());
+            return await Backchannel.SendAsync(request, Context.RequestAborted);
+        }
+
+        private async Task<RequestToken> ObtainRequestTokenAsync(string callBackUri, AuthenticationProperties properties)
+        {
+            Logger.ObtainRequestToken();
 
-            var response = await Backchannel.SendAsync(request, Context.RequestAborted);
+            var response = await ExecuteRequestAsync(RequestTokenEndpoint, HttpMethod.Post, null, new Dictionary<string, string>() { { "oauth_callback", callBackUri } });
```

Comment:
You can drop the null param, it has a default, and use a named parameter for the last param. The same goes for the other calls to ExecuteRequestAsync.